src/operator/tensor/elemwise_unary_op_trig.cc

```diff
@@ -188,7 +188,31 @@ The storage type of ``arcsin`` output depends upon the input storage type:
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{ "_backward_arcsin" });
 
 MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU_DR(_backward_arcsin,
-                                                  unary_bwd<mshadow_op::arcsin_grad>);
+                                                  unary_bwd<mshadow_op::arcsin_grad>)
+.set_attr<nnvm::FGradient>("FGradient",
+    [](const nnvm::NodePtr& n, const std::vector<nnvm::NodeEntry>& ograds) {
+      // ograds[0]: head_grad_grads (dL/dxgrad)
+      // inputs[0]: dL/dy
+      // inputs[1]: x (ElemwiseGradUseIn)
+      // f(x) = arcsin(x)
+      // n: f'(x) = 1/(1-x^2)^1/2
+      // f''(x) = f'(x) * x/(1-x^2)
+      // Note: x/(1-x^2) = x * f'(x)^2
+      auto dydx = n->inputs[0];
+      auto x = n->inputs[1];
+      auto dydx_mul_grad_x = nnvm::NodeEntry{n};
+      auto op = mxnet::util::NodeOpGen{n};
+
+      auto grad_x = op.div(dydx_mul_grad_x, dydx);
+      auto grad_x_square = op.square(grad_x);
+      auto grad_x_square_mul_x = op.mul(grad_x_square, x);
+      auto grad_grad_x = op.mul(dydx_mul_grad_x, grad_x_square_mul_x);
+
+      std::vector<nnvm::NodeEntry> ret;
+      ret.emplace_back(op.mul(ograds[0], grad_x));
```

Comment:
if the first input is y_grad or dL/dy, this gradient should be dL/(dy*dx)  ?
didn't we have the convention of x_grad instead of grad_x?

Reply:
Sorry for the late reply.

> if the first input is y_grad or dL/dy, this gradient should be dL/(dy*dx) ?

I am not sure of `dL` part because we don't really use it in computing the loss function. 

> didn't we have the convention of x_grad instead of grad_x?

Oops. Thanks. Was a old PR. Will update the names.

Reply:
Sure, l guess calling it x_grad_y_grad is fine. Sorry CI is flaky now.

Reply:
`x_grad_y_grad` we are not naming that particular variable anywhere. Or am I confusing something?

Reply:
@larroy Can you elaborate on what you meant with `x_grad_y_grad`. I am slightly confused. Thanks.